cache/file/file_test.go

```diff
@@ -0,0 +1,108 @@
+package file
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/pierrre/imageserver"
+	imageserver_cache "github.com/pierrre/imageserver/cache"
+	cachetest "github.com/pierrre/imageserver/cache/_test"
+	"github.com/pierrre/imageserver/testdata"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+var _ imageserver_cache.Cache = &Cache{}
+var testDirPath string
+
+func TestMain(m *testing.M) {
+	os.Exit(realMain(m))
+}
+
+func realMain(m *testing.M) int {
+	path, err := ioutil.TempDir("", "filecache")
+	if err != nil {
+		return 1
+	}
+	testDirPath = path
+	defer func() {
+		_ = os.RemoveAll(testDirPath)
+	}()
+	return m.Run()
+}
+
+func TestGetSet(t *testing.T) {
+	cache := newTestCache(t)
+	cachetest.TestGetSet(t, cache)
+}
+
+func TestGetMiss(t *testing.T) {
+	cache := newTestCache(t)
+	cachetest.TestGetMiss(t, cache)
+}
+
+func TestPathIsNotSet(t *testing.T) {
+	cache := &Cache{Path: "", Mutex: sync.RWMutex{}}
```

Comment:
You don't need to initialize the mutex field.
The default value is already initialized.